File: src/org/usfirst/frc5293/commands/teleop/control/DrivetrainControl.java
package org.usfirst.frc5293.commands.teleop.control;

import org.usfirst.frc5293.Subsystems;
import org.usfirst.frc5293.commands.util.ContinuousCommand;
import org.usfirst.frc5293.translations.util.DrivingState;
import org.usfirst.frc5293.translations.driving.JoystickDriveEngine;

public class DrivetrainControl extends ContinuousCommand {

    public DrivetrainControl() {
        requires(Subsystems.getDrivetrain());
    }

    @Override
    protected void execute() {
        drive(JoystickDriveEngine.getInstance().getResult());
    }

    private void drive(DrivingState state) {
        Subsystems.getDrivetrain().drive(state.power, state.rotation);
    }

    @Override
    protected void end() {
    	Subsystems.getDrivetrain().stop();
    }
}

File: src/org/usfirst/frc5293/commands/teleop/control/DrivetrainControl.kt
package org.usfirst.frc5293.commands.teleop.control

import org.usfirst.frc5293.Input
import org.usfirst.frc5293.Subsystems
import org.usfirst.frc5293.commands.util.EmptyCommand

class DrivetrainControl : EmptyCommand() {

    init {
        requires(Subsystems.drivetrain)
    }

    override fun execute() {
        Subsystems.drivetrain.driveArcade(
                power = Input.drivetrain.powerJoystick.y,
                rotation = Input.drivetrain.rotationJoystick.twist
        )
    }

//    private fun drive(state: TankDrivingState) {
//        Subsystems.drivetrain.driveTank(state.left, state.right)
//    }

    override fun end() {
        Subsystems.drivetrain.stop()
    }
}
